practices/Ramon-OC/practica01/practica01.py

```diff
@@ -0,0 +1,332 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.16.1
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# + [markdown] id="ioqX3obg5WcJ"
+# # Práctica 1: Niveles lingüísticos I
+# **Fecha de entrega: Domingo 3 de Marzo 2024 11:59pm**
+
+# + id="dfCkH58988vq"
+import requests as r
+from difflib import get_close_matches
+
+response = r.get("https://raw.githubusercontent.com/open-dict-data/ipa-dict/master/data/en_US.txt")
+
+
+# + [markdown] id="cJMkPF06jJJp"
+# ### Obteniendo el corpus
+
+# + id="34e980ae-c902-4f62-b1d2-a41b9e9939e3"
+def response_to_dict(ipa_list: list) -> dict:
+    """Parse to dict the list of word-IPA
+
+    Each element of text have the format:
+    [WORD][TAB][IPA]
+
+    Parameters
+    ----------
+    ipa_list: list
+        List with each row of ipa-dict raw dataset file
+
+    Returns
+    -------
+    dict:
+        A dictionary with the word as key and the phonetic
+        representation as value
+    """
+    result = {}
+    for item in ipa_list:
+       item_list = item.split("\t")
+       result[item_list[0]] = item_list[1]
+    return result
+
+
+# + id="8e446564-cb5a-41a0-ac9a-e26f91e25273"
+def get_ipa_dict(iso_lang: str) -> dict:
+    """Get ipa-dict file from Github
+
+    Parameters:
+    -----------
+    iso_lang:
+        Language as iso code
+
+    Results:
+    --------
+    dict:
+        Dictionary with words as keys and phonetic representation
+        as values for a given lang code
+    """
+    print(f"Downloading {iso_lang}", end=" ")
+    response = r.get(f"https://raw.githubusercontent.com/open-dict-data/ipa-dict/master/data/{iso_lang}.txt")
+    raw_data = response.text.split("\n")
+    print(f"status:{response.status_code}")
+    return response_to_dict(raw_data[:-1])
+
+
+# + id="3vfeGyqYkI9V"
+def query_ipa_transcriptions(word: str, dataset: dict) -> list[str]:
+    """Search for a word in an IPA phonetics dict
+
+    Given a word this function return the IPA transcriptions
+
+    Parameters:
+    -----------
+    word: str
+        A word to search in the dataset
+    dataset: dict
+        A dataset for a given language code
+
+    Returns
+    -------
+    list[str]:
+        List with posible transcriptions if any,
+        else a list with the string "NOT FOUND"
+    """
+    return dataset.get(word.lower(), "NOT FOUND").split(", ")
+
+
+# + id="65a1e3df"
+def word_IPA(searchWord: str, dataset: dict) -> str:
```

Comment:
Recomiendo definir todas las funciones en minúsculas para que cumpla con el estandar de python

```suggestion
def get_word_ipa(searchWord: str, dataset: dict) -> str:
```